### PR TITLE
implement secret handling for workspace_service_account table

### PR DESCRIPTION
**Pull request by subodh1810**
Issue : https://github.com/airbytehq/airbyte/issues/11968

### CODE REVIEW

File: airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/SecretsRepositoryReaderTest.java [hunk 1]
```diff
@@ -116,4 +122,111 @@ void testDumpConfigsWithSecrets() throws IOException {
...
 
+  @Test
+  public void testReadingServiceAccount() throws JsonValidationException, ConfigNotFoundException, IOException {
+    final ConfigRepository configRepository = mock(ConfigRepository.class);
+    final SecretPersistence secretPersistence = mock(SecretPersistence.class);
+    final RealSecretsHydrator realSecretsHydrator = new RealSecretsHydrator(secretPersistence);
+    final SecretsRepositoryReader secretsRepositoryReader =
+        spy(new SecretsRepositoryReader(configRepository, realSecretsHydrator));
+
+    final UUID workspaceId = UUID.fromString("13fb9a84-6bfa-4801-8f5e-ce717677babf");
```
Comment (tuliren) [inline airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/SecretsRepositoryReaderTest.java:133]: Many of the UUIDs and constants are reused in the two test files, like this workspace ID, the service account ID, and HMAC key. They can be moved to `MockData.java` for clarity.

**Review by tuliren — APPROVED** [review]

**Review by edgao — APPROVED** [review]

**Review by cgardens — CHANGES_REQUESTED** [review]
i think the approach makes sense! nice work. 

mostly comments on clarity. i want to be really picky about that here since secrets are involved.

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretsHelpers.java [hunk 2]
```diff
@@ -336,16 +336,18 @@ private static JsonNode getFieldOrEmptyNode(final JsonNode node, final int field
    * @param secretPersistence storage layer for secrets
    * @param coordinate reference to a secret in the persistence
    * @throws RuntimeException when a secret at that coordinate is not available in the persistence
-   * @return a json text node containing the secret value
+   * @return a json text node containing the secret value or a JSON
    */
-  private static TextNode getOrThrowSecretValueNode(final ReadOnlySecretPersistence secretPersistence, final SecretCoordinate coordinate) {
+  private static JsonNode getOrThrowSecretValueNode(final ReadOnlySecretPersistence secretPersistence,
+                                                    final SecretCoordinate coordinate,
+                                                    final boolean asTextNode) {
```
[the remark this answers is not in the document]
Comment (tuliren) [reply airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretsHelpers.java:343]: Alternatively, `getOrThrowSecretValueNode` is called two times, once to get the text node, and once to get the json node. So this helper method is not that useful. It's also fine to just remove it.

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretsHelpers.java [hunk 3]
```diff
@@ -336,16 +336,16 @@ private static JsonNode getFieldOrEmptyNode(final JsonNode node, final int field
    * @param secretPersistence storage layer for secrets
    * @param coordinate reference to a secret in the persistence
    * @throws RuntimeException when a secret at that coordinate is not available in the persistence
-   * @return a json text node containing the secret value
+   * @return a json text node containing the secret value or a JSON
    */
-  private static TextNode getOrThrowSecretValueNode(final ReadOnlySecretPersistence secretPersistence, final SecretCoordinate coordinate) {
+  private static String getOrThrowSecretValueNode(final ReadOnlySecretPersistence secretPersistence,
```
Comment (cgardens) [inline airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretsHelpers.java:341]: `getOrThrowSecretValueNode` => `getOrThrowSecretValue` or even `getOrThrowSecret`. It's not a "node" anymore, right?

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryWriter.java [hunk 4]
```diff
@@ -264,4 +266,53 @@ public void replaceAllConfigs(final Map<AirbyteConfig, Stream<?>> configs, final
...
+                longLivedSecretPersistence.get(),
+                workspaceServiceAccount.getWorkspaceId(),
+                UUID::randomUUID,
+                optionalWorkspaceServiceAccount.map(WorkspaceServiceAccount::getJsonCredential).orElse(null),
+                "json");
+        longLivedSecretPersistence.get().write(jsonCredSecretCoordinateToPayload.secretCoordinate(), jsonCredSecretCoordinateToPayload.payload());
+        clonedWorkspaceServiceAccount.setJsonCredential(jsonCredSecretCoordinateToPayload.secretCoordinateForDB());
+      }
+
+      if (workspaceServiceAccount.getHmacKey() != null) {
```
[the remark this answers is not in the document]
Comment (cgardens) [reply airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryWriter.java:296]: thanks. is all of this specific to a GCP service account (as opposed to an AWS one?)? If they are different how will we tell in the future? As long as we have answer to this, then all good. If it GCP specific now, worth adding a comment mentioning that as well and that we plan for it to evolve.

**Review by cgardens — APPROVED** [review]
Thanks! LGTM.

**Pull request merged**